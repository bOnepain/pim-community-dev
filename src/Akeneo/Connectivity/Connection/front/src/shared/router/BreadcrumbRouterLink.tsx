import React, {ComponentPropsWithoutRef, useContext} from 'react';
import {BreadcrumbItem} from '../../common';
import {RouterContext} from './index';

interface Props {
    route: string;
}

export const BreadcrumbRouterLink = ({
    children: label,
    route,
}: Props & ComponentPropsWithoutRef<typeof BreadcrumbItem>) => {
    const router = useContext(RouterContext);

    return <BreadcrumbItem onClick={() => router.redirect(router.generate(route))}>{label}</BreadcrumbItem>;
};
