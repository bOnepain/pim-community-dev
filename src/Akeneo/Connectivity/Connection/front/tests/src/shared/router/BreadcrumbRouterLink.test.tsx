import {act, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as React from 'react';
import {create} from 'react-test-renderer';
import {BreadcrumbRouterLink, RouterContext} from '@src/shared/router';

describe('BreadcrumbRouterLink', () => {
    it('should render', () => {
        const component = create(<BreadcrumbRouterLink route='a_route'>content</BreadcrumbRouterLink>);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should redirect to a route', async () => {
        const router = {
            generate: jest.fn().mockReturnValue('/an_url'),
            redirect: jest.fn(),
        };

        const {getByText} = render(
            <RouterContext.Provider value={router}>
                <BreadcrumbRouterLink route='a_route'>content</BreadcrumbRouterLink>
            </RouterContext.Provider>
        );

        await act(async () => {
            userEvent.click(getByText('content'));

            return Promise.resolve();
        });

        expect(router.generate).toBeCalledWith('a_route');
        expect(router.redirect).toBeCalledWith('/an_url');
    });
});
