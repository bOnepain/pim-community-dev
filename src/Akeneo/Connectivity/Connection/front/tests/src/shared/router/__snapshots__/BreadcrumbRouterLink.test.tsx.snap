// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbRouterLink should render 1`] = `
<span
  className="AknBreadcrumb-item AknBreadcrumb-item--routable"
  onClick={[Function]}
>
  content
</span>
`;
